File: Examples/AppWithFirebase/app/src/main/java/com/optimove/sdk/demo/screen_visit/ScreenVisitFragment.java
package com.optimove.sdk.demo.screen_visit;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.textfield.TextInputEditText;
import com.optimove.sdk.demo.R;
import com.optimove.sdk.optimove_sdk.main.Optimove;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class ScreenVisitFragment extends Fragment {

  private TextInputEditText titleInput;
  private TextInputEditText categoryInput;

  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_screen_visit, container, false);
    titleInput = view.findViewById(R.id.pageTitleInput);
    categoryInput = view.findViewById(R.id.categoryInput);
    view.findViewById(R.id.setScreenVisitButton).setOnClickListener(v -> setScreenVisit());
    return view;
  }

  public void setScreenVisit() {
    String title = titleInput.getText().toString();
    String category = categoryInput.getText().toString();

    Optimove.getInstance().reportScreenVisit(title, category);
  }
}

File: Examples/Kotlin/app/src/main/java/com/optimove/sdk/demo/screen_visit/ScreenVisitFragment.kt
package com.optimove.sdk.demo.screen_visit

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.google.android.material.textfield.TextInputEditText
import com.optimove.sdk.demo.R
import com.optimove.sdk.optimove_sdk.main.Optimove
import androidx.fragment.app.Fragment

class ScreenVisitFragment : Fragment() {

    private var urlInput: TextInputEditText? = null
    private var titleInput: TextInputEditText? = null
    private var categoryInput: TextInputEditText? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_screen_visit, container, false)
        urlInput = view.findViewById(R.id.customUrlInput)
        titleInput = view.findViewById(R.id.pageTitleInput)
        categoryInput = view.findViewById(R.id.categoryInput)
        view.findViewById<View>(R.id.setScreenVisitButton).setOnClickListener { v -> setScreenVisit() }
        return view
    }

    fun setScreenVisit() {
        val url = urlInput!!.text!!.toString()
        val title = titleInput!!.text!!.toString()
        val category = categoryInput!!.text!!.toString()

        Optimove.getInstance().setScreenVisit(url, title, category)
    }
}
